model_analyzer/result/run_config_measurement.py

```diff
@@ -93,6 +96,17 @@ def set_model_config_weighting(self, model_config_weights):
             for model_config_weight in model_config_weights
         ]
 
+    def set_model_config_constraints(self, model_config_constraints):
+        """
+
+        Parameters
+        ----------
+            model_config_constraints: dict (single model) or list of dicts (multi-model)
+            Used to determine if an ModelConfigMeasurement passes or fails
+        """
+        self._model_config_constraints = [model_config_constraints] if type(
+            model_config_constraints) is dict else model_config_constraints
```

Comment:
I think method should stick with a single input type, and it should be the caller's job to match the type.

Reply:
Understood. I had just copied what was needed when setting constraints for the RunConfigResult class, but yes I agree this is not needed here. I will just expect a list.